libraries/AP_RCProtocol/AP_RCProtocol_Backend.cpp

```diff
@@ -76,28 +76,28 @@ void AP_RCProtocol_Backend::add_input(uint8_t num_values, uint16_t *values, bool
 }
 
 
-// decode channels from the standard 11bit format (used by CRSF and SBUS)
+/*
+  decode channels from the standard 11bit format (used by CRSF, SBUS, FPort and FPort2)
+  must be used on multiples of 8 channels
+*/
 void AP_RCProtocol_Backend::decode_11bit_channels(const uint8_t* data, uint8_t nchannels, uint16_t *values, uint16_t mult, uint16_t div, uint16_t offset)
 {
 #define CHANNEL_SCALE(x) ((int32_t(x) * mult) / div + offset)
+    while (nchannels >= 8) {
+        const Channels11Bit_8Chan* channels = (const Channels11Bit_8Chan*)data;
```

Comment:
Why bother with this chunking at all?  Why not just iterate to `nchannels`, taking a uint32_t off `data` each time?

Reply:
its in 11 bit chunks, using the bitfield method needs to go in multiples like this